client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchDetails.tsx

```diff
@@ -0,0 +1,600 @@
+  resultTallySheets: IBatchResultTallySheet[]
+}
+
+const VOTE_TOTALS_TAB_ID = 'vote-totals'
+
+function defaultResultTallySheetName(index: number) {
+  return `Sheet ${index + 1}`
+}
+
+function constructEmptyResultTallySheet(index: number): IBatchResultTallySheet {
+  return {
+    name: defaultResultTallySheetName(index),
+    results: {},
+  }
+}
+
+interface IBatchDetailsProps {
+  areResultsFinalized: boolean
+  batch: IBatch
+  contest: IContest
+  saveBatchResults: (
+    resultTallySheets: IBatchResultTallySheet[]
+  ) => Promise<void>
+  setAreChangesUnsaved: (areChangesUnsaved: boolean) => void
+
+  // Require a key to ensure that the form within this component resets when a different batch is
+  // selected
+  key: string // eslint-disable-line react/no-unused-prop-types
+}
+
+const BatchDetails: React.FC<IBatchDetailsProps> = ({
+  areResultsFinalized,
+  batch,
+  contest,
+  saveBatchResults,
+  setAreChangesUnsaved,
+}) => {
+  const [selectedTabId, setSelectedTabId] = useState(VOTE_TOTALS_TAB_ID)
+  const [isEditing, setIsEditing] = useState(false)
+  const [
+    areAdditionalSheetActionsOpen,
+    setAreAdditionalSheetActionsOpen,
+  ] = useState(false)
+  const [isRenamingSheet, setIsRenamingSheet] = useState(false)
+  const [draftSheetName, setDraftSheetName] = useState('')
+  const [
+    autoSelectLastTabOnceNumSheetsIs,
+    setAutoSelectLastTabOnceNumSheetsIs,
+  ] = useState<number>()
+
+  const initialResultTallySheets = batch.resultTallySheets.length
+    ? batch.resultTallySheets
+    : [constructEmptyResultTallySheet(0)]
+  const formMethods = useForm<IBatchRoundDataEntryFormState>({
+    defaultValues: {
+      resultTallySheets: initialResultTallySheets,
+    },
+  })
+  const {
+    control,
+    formState,
+    getValues,
+    handleSubmit,
+    reset,
+    setValue,
+    watch,
+  } = formMethods
+  // Important gotcha! You have to access properties on the formState to subscribe to it:
+  // https://github.com/react-hook-form/react-hook-form/issues/9002
+  const { isDirty, isSubmitting } = formState
+  const sheetFieldArrayMethods = useFieldArray<IBatchResultTallySheet>({
+    control,
+    name: 'resultTallySheets',
+  })
+  const sheetFields = sheetFieldArrayMethods.fields
+
+  const currentSheetIndex = sheetFields.findIndex(s => s.id === selectedTabId)
+
+  const selectVoteTotalsTab = () => {
+    setSelectedTabId(VOTE_TOTALS_TAB_ID)
+  }
+
+  const selectTabWithSheetIndex = (sheetIndex: number) => {
+    setSelectedTabId(sheetFields[sheetIndex].id || '')
+  }
+
+  // Auto-select new sheets
+  useEffect(() => {
+    if (
+      autoSelectLastTabOnceNumSheetsIs !== undefined &&
+      sheetFields.length === autoSelectLastTabOnceNumSheetsIs
+    ) {
+      selectTabWithSheetIndex(sheetFields.length - 1)
+      setAutoSelectLastTabOnceNumSheetsIs(undefined)
+    }
+  }, [
+    autoSelectLastTabOnceNumSheetsIs,
+    selectTabWithSheetIndex,
+    setAutoSelectLastTabOnceNumSheetsIs,
+    sheetFields,
+  ])
```

Comment:
Not my favorite part of this code 😅

I know `autoSelectLastTabOnceNumSheetsIs` is a convoluted name. The challenge here is that we use `useFieldArray` sheet field IDs as tab IDs. And when a new sheet is created, we don't have the new sheet field ID right away.

In earlier iterations of this code, I tried used other fields that we know upon sheet creation as tab IDs, e.g. sheet indexes and sheet names, but both of those created other bigger headaches, like tabs not updating properly